docs/build/apps/dapp-frontend.mdx

```diff
@@ -1,561 +1,536 @@
 
-Once it's installed, open it up by clicking the extension icon. If this is your first time using Freighter, you will need to create a new wallet. Go through the prompts to create a password and save your recovery passphrase.
+  for (const wasm of wasmFiles) {
+    await deploy(wasm);
+  }
+}
+```
 
-Go to Settings (the gear icon) → Preferences and toggle the switch to Enable Experimental Mode. Then go back to its home screen and select "Test Net" from the top-right dropdown. Finally, if it shows the message that your Stellar address is not funded, go ahead and click the "Fund with Friendbot" button.
+The `deploy()` function will get the contract ID from the CLI call, and add the contract name, wasm file path, and contract ID to the `smartContracts[]` array.
 
-Now you're all set up to use Freighter as a user, and you can add it to your app.
+In this example, we only use one smart contract, but it’s not uncommon to use multiple smart contracts in a dapp, so the template supports the use of multiple contracts.
 
-### Add the StellarWalletsKit and set it up
+##### Create bindings
 
-Even though we're using Freighter to test our app, there are more wallets that support signing smart contract transactions. To make their integration easier, we are using the `StellarWalletsKit` library which allows us support all Stellar Wallets with a single library.
+The Stellar CLI has a convenient command to create an NPM package that makes it easy to call smart contract functions from a JavaScript/TypeScript-based frontend. We call the package “bindings” because that’s what it does: it binds the contract and the frontend together.
 
-To install this kit we are going to include the next package:
+As with the contract build functions, the binding function is also capable of handling multiple contracts, so there’s a function for creating the binding package for a contract (`bind()`) and a function that calls `bind()` for each contract (`bindAll()`).
 
-```shell
-npm install @creit.tech/stellar-wallets-kit
+```javascript
+function bind({ alias, wasm, contractid }) {
+  // Create bindings for a deployed contract
+  execSync(
+    `stellar contract bindings typescript --contract-id ${contractid} --output-dir ${dirname}/packages/${alias} --overwrite`,
+  );
+  // Build the package
+  execSync(`(cd ${dirname}/packages/${alias} && npm i && npm run build)`);
+}
+
+async function bindAll() {
+  // Bind all deployed contracts
+  for (const contract of smartContracts) {
+    await bind(contract);
+  }
+}
 ```
 
-With the package installed, we are going to create a new simple file where our instantiated kit and simple state will be located. Create the file `src/stellar-wallets-kit.ts` and paste this:
+The `bindAll()` function iterates the `smartContracts[]` array. The reason for not just using the array of wasms, like in the `deployAll()` function, is that we need the contract ID to generate the bindings.
```

Comment:
```suggestion
The `bindAll()` function iterates the `smartContracts[]` array. The reason for not just using the array of wasms, like in the `deployAll()` function, is that we need the contract ID to invoke the generated bindings functions on the network.
```

you can actually generate bindings from a Wasm file, but this will leave you with a client instance that requires you to specify a address